modernizer-maven-plugin/src/main/java/org/gaul/modernizer_maven_plugin/Modernizer.java

```diff
@@ -250,6 +273,19 @@ private boolean ignoreClass() {
         return false;
     }
 
+    private boolean ignoreMethod(String methodName, String methodDescriptor) {
+        String methodParams = methodDescriptor.substring(
+            methodDescriptor.indexOf('(') + 1,
+            methodDescriptor.indexOf(')'))
+            .replace('$', '/');
+        String methodDescription = className.replace("$", "\\$") + "," +
+            methodName + "," + methodParams;
+        if (ignoreMethods.contains(methodDescription)) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
Rather than needing to parse the method descriptor here, could we construct the full method descriptor in the annotation processor? Also, it seems like ASM has some utilities for parsing these method descriptors which might handle more edge cases for us: http://asm.ow2.org/asm50/javadoc/user/org/objectweb/asm/Type.html

Reply:
We can construct the descriptor in the processor itself - the only extra info we need is the return type. Once we have the whole descriptor here, we can just form the `methodDescription` string and check if this is contained in `ignoreMethods` right? Do we still need to parse the descriptor? What type of edge cases would these utilities handle?

Reply:
I think I had initially been worried about adding ASM as a dep to the processor, but it should be fine.